saleor/order/actions.py

```diff
@@ -1463,51 +1466,44 @@ def _process_refund(
         if refund_shipping_costs:
             amount += order.shipping_price_gross_amount
     if amount:
-        msg = "The refund operation is not available yet."
-        code = OrderErrorCode.CANNOT_REFUND.value
-        try:
-            if not payment.gateway:
-                code = OrderErrorCode.MISSING_PAYMENT_ACTION_REQUEST_WEBHOOK.value
-                msg = (
-                    "No app or plugin is configured to handle payment action requests."
-                )
-                amount = min(payment.captured_value, amount)
-                request_refund_action(
-                    payment,
-                    manager,
-                    refund_value=amount,
-                    channel_slug=order.channel.slug,
+        if transactions:
+            # With current mutation's inputs, we are able to process only single
+            # transaction. This can be changed when we will provide an interface
+            # to provide list of transactions.
+            transaction_item = transactions[-1]
+            amount = min(transaction_item.captured_value, amount)
+            request_refund_action(
+                transaction_item,
+                manager,
+                refund_value=amount,
+                channel_slug=order.channel.slug,
+                user=user,
+                app=app,
+            )
```

Comment:
Should we create any event here?

Reply:
It is handled by `request_refund_action`. It calls `event_payment_refund_requested`